packages/react-router/src/transformer.ts

```diff
@@ -162,9 +160,14 @@ export type TransformerParse<T, TSerializable> = T extends TSerializable
     ? ReadableStream
     : { [K in keyof T]: TransformerParse<T[K], TSerializable> }
 
+export type DefaultSerializeable = Date | undefined | Error | FormData
```

Comment:
to avoid this being missed in future, it *could* be extracted directly from the transformers array:
```ts
type DefaultTransformers = (typeof transformers)[number]

export type DefaultSerializeable = {
  [K in DefaultTransformers['key']]: Extract<
    DefaultTransformers,
    { key: K }
  > extends { stringifyCondition: (value: any) => value is infer T }
    ? T
    : never
}[DefaultTransformers['key']]
```
but it makes the compiled output all icky and i suspect the performance will be impacted slightly, so it's a tradeoff

Reply:
you could also flip it, and check the transformers array *against* the desired types:
```ts
type DefaultSerializeableByFrequency = [undefined, Date, Error, FormData]

type TransformersFor<T extends Array<any>> = {
  [K in keyof T]: { stringifyCondition: (value: any) => value is T[K] }
}

const transformers = [...] as const satisfies TransformersFor<DefaultSerializeableByFrequency>

export type DefaultSerializeable = DefaultSerializeableByFrequency[number]
```
this isn't quite as nice for DX since it's manually specified instead of inferred, but the build output will be nicer (and it'll still complain if there's a mismatch)

Reply:
Could you instead just add a comment at the top of the array of transformers asking any future adventurers' to add the type into this union?